dix/req.c

```diff
@@ -2270,11 +2332,25 @@ static int dix_cas_rops_alloc(struct m0_dix_req *req)
 		cas_rop->crp_cur_key = 0;
 	} m0_tl_endfor;
 
+	if (dtx == NULL)
+		goto end;
+
+	for (i = 0; i < skipped_sdevs_max && rc == 0; i++) {
+		if (skipped_sdevs[i] != invalid_sdev_id) {
+			cas_svc = pc->pc_dev2svc[skipped_sdevs[i]].pds_ctx;
+			M0_ASSERT(cas_svc->sc_type == M0_CST_CAS);
+			M0_ASSERT(pa_idx < pa_nr);
+			rc = m0_dtx0_fid_assign(dtx, pa_idx,
+						&cas_svc->sc_fid);
+			pa_idx++;
+		}
+	}
 end:
 	if (rc != 0) {
 		dix_cas_rops_fini(&rop->dg_cas_reqs);
 		return M0_ERR(rc);
```

Comment:
Looks like we will leak `skipped_sdevs` here.

Reply:
Right, fixed.